connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -237,7 +246,15 @@ object KustoWriter {
         val blobPath = blobUri + sas
         val blobSourceInfo = new BlobSourceInfo(blobPath, size)
 
-        partitionsResults.add(PartitionResult(ingestClient.ingestFromBlob(blobSourceInfo, props), partitionId))
+        val res: IngestionResult = try {
+          KDSU.logInfo(myName, s"Queued blob for ingestion in partition ${TaskContext.getPartitionId} for requestId: '$requestId}")
+          ingestClient.ingestFromBlob(blobSourceInfo, props)
+        } catch {
+          // Catch only IngestionServiceException cause it can be due to transient storage issue
+          case _: IngestionServiceException =>
+            ingestClient.ingestFromBlob(blobSourceInfo, props)
```

Comment:
Just making sure this was intentional - this is outside the try, so if it throws, it won't be added to `partitionsResults`.

Reply:
It shouldnt - if we fail here - we fail everrything